plugins/graphiql/src/components/GraphiQLPage/GraphiQLPage.tsx

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { FC } from 'react';
+import {
+  Content,
+  Header,
+  HeaderLabel,
+  Page,
+  Progress,
+  pageTheme,
+  useApi,
+} from '@backstage/core';
+import { useAsync } from 'react-use';
+import 'graphiql/graphiql.css';
+import { graphQlBrowseApiRef } from 'lib/api';
+import { GraphiQLBrowser } from 'components';
+import { Typography } from '@material-ui/core';
+
+export const GraphiQLPage: FC<{}> = () => {
+  const graphQlBrowseApi = useApi(graphQlBrowseApiRef);
+  const endpoints = useAsync(() => graphQlBrowseApi.getEndpoints());
+
+  let content: JSX.Element;
+
+  if (endpoints.loading) {
+    content = (
+      <Content>
+        <Progress />
+      </Content>
+    );
+  } else if (endpoints.error) {
+    content = (
+      <Content>
+        <Typography variant="h4" color="error">
+          Failed to load GraphQL endpoints, {endpoints.error}
+          <br />
+          We also need a proper error component
+        </Typography>
+      </Content>
+    );
+  } else {
+    content = (
+      <Content noPadding>
+        <GraphiQLBrowser endpoints={endpoints.value!} />
+      </Content>
+    );
+  }
+
+  return (
+    <Page theme={pageTheme.tool}>
+      <Header title="GraphiQL">
+        <HeaderLabel label="Owner" value="Spotify" />
+        <HeaderLabel label="Lifecycle" value="Alpha" />
```

Comment:
I wonder if the these header labels and their values should be hardcoded in the plugin. Perhaps this is just the beginning, but ideally we might be storing these as a plugin property. (I'm curious how is it done for internal Backstage). 🙏

Reply:
For Tools like this they are hard coded, but if you are looking at a Backstage page that shows e.g. a Service, then we fetch the Service over via GraphQL and show it in the header (together with a few other attributes).

Reply:
Yep, internally we have some more metadata tied to a plugin than just the ID, haven't decided exactly on how to expose that yet though. Agree that we should likely figure out a better way to handle them.